types/validator_set.go

```diff
@@ -399,13 +399,14 @@ func verifyUpdates(
 			// Updated validator, add the difference in power to the total.
 			updatedTotalVotingPower += valUpdate.VotingPower - val.VotingPower
 		}
-		overflow := updatedTotalVotingPower > MaxTotalVotingPower
-		if overflow {
-			err = fmt.Errorf(
-				"failed to add/update validator %v, total voting power would exceed the max allowed %v",
-				valUpdate, MaxTotalVotingPower)
-			return 0, 0, err
-		}
+	}
+
+	overflow := updatedTotalVotingPower > MaxTotalVotingPower
```

Comment:
is it possible the voting power actually overflowed the int64 during the for loop tho?

Reply:
Yes, this will overflow and panic. Here is the stacktrace:
goroutine 54 [running]:
testing.tRunner.func1(0xc000556100)
	/usr/local/Cellar/go/1.12.5/libexec/src/testing/testing.go:830 +0x392
panic(0x168c320, 0xc0000cd230)
	/usr/local/Cellar/go/1.12.5/libexec/src/runtime/panic.go:522 +0x1b5
github.com/tendermint/tendermint/types.(*ValidatorSet).updateTotalVotingPower(0xc0004af770)
	/Users/ancaz/go/src/github.com/tendermint/tendermint/types/validator_set.go:268 +0x17e
github.com/tendermint/tendermint/types.(*ValidatorSet).updateWithChangeSet(0xc0004af770, 0xc0004afe00, 0x5, 0x5, 0x1, 0x5, 0x0)
	/Users/ancaz/go/src/github.com/tendermint/tendermint/types/validator_set.go:571 +0x20a
github.com/tendermint/tendermint/types.(*ValidatorSet).UpdateWithChangeSet(...)
	/Users/ancaz/go/src/github.com/tendermint/tendermint/types/validator_set.go:593
github.com/tendermint/tendermint/types.applyChangesToValSet(0xc000556100, 0xc0004af770, 0xc000063d30, 0x3, 0x3)
	/Users/ancaz/go/src/github.com/tendermint/tendermint/types/validator_set_test.go:1112 +0x17d
github.com/tendermint/tendermint/types.verifyValSetUpdatePriorityOrder(0xc000556100, 0xc0004af770, 0xc000476180, 0x5, 0x5, 0x1eab140, 0x0, 0x0, 0xc000476200, 0x5, ...)
	/Users/ancaz/go/src/github.com/tendermint/tendermint/types/validator_set_test.go:1160 +0x1df
github.com/tendermint/tendermint/types.TestValSetUpdatePriorityOrderTests(0xc000556100)
	/Users/ancaz/go/src/github.com/tendermint/tendermint/types/validator_set_test.go:1148 +0x2f7
testing.tRunner(0xc000556100, 0x18ccdf0)
	/usr/local/Cellar/go/1.12.5/libexec/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.12.5/libexec/src/testing/testing.go:916 +0x35a